### PR TITLE
[4.0] Fixing switcher label #26749

### PR DESCRIPTION
**Pull request by infograf768**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/26749

### Summary of Changes
Taking into account both int and string possibility to display No and Hide to be B/C

### Testing Instructions
Display Global Configuration

### Before patch
Screen Shot 2019-11-02 at 08 20 22

### After patch
Screen Shot 2019-11-02 at 08 17 57

### CODE REVIEW

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on fe97446fcb35fdde04555e868bb9957a9dd4b2f3<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26908">issues.joomla.org/tracker/joomla-cms/26908</a>.</sub>

**Comment by rajnish-dargan** [comment]
I have tested this item :white_check_mark: successfully on fe97446fcb35fdde04555e868bb9957a9dd4b2f3<br /><br />#pbf19 #test1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26908">issues.joomla.org/tracker/joomla-cms/26908</a>.</sub>

**Comment by vaibhavsTekdi** [comment]
I have tested this item :white_check_mark: successfully on fe97446fcb35fdde04555e868bb9957a9dd4b2f3<br /><br />#pbf19 #test2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26908">issues.joomla.org/tracker/joomla-cms/26908</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26908">issues.joomla.org/tracker/joomla-cms/26908</a>.</sub>

**Comment by SharkyKZ** [comment]
Doesn't work with float-like values.

**Comment by infograf768** [comment]
@SharkyKZ 
> Doesn't work with float-like values.

I do not understand. Please explain a case.

It never works for any other option value than 0/1 in the xml

**Comment by Quy** [comment]
The issue is that `$value` can be empty, thus, not equal to `0`. The following is possible solution.

```
$value		= $value === '' ? '0' : $value ;
```

**Comment by Quy** [comment]
Please view `configuration.php`.  Look for settings with `true` and `false` values.
Under `System > System Information > Configuration File`, `true` and `false` are displayed as `1` as a string and empty string respectively.

Therefore, you only need to check for empty string (`false`) and assign it `0` as a string which is a simpler fix.

```
		$value		= $value === '' ? '0' : $value ;
		$checked	= ((string) $option->value == $value) ? 'checked="checked"' : '';
		$active		= ((string) $option->value == $value) ? 'class="active"' : '';
```

**Comment by infograf768** [comment]
@Quy Done
Took off RTC until tested again.
@alikon @SharkyKZ

File: layouts/joomla/form/field/radio/switcher.php [hunk 1]
```diff
@@ -69,12 +69,13 @@
 	<?php foreach ($options as $i => $option) : ?>
 		<?php
 		// Initialize some option attributes.
-		$checked	= ((string) $option->value == $value) ? 'checked="checked"' : '';
-		$active		= ((string) $option->value == $value) ? 'class="active"' : '';
-		$oid		= $id . $i;
-		$ovalue		= htmlspecialchars($option->value, ENT_COMPAT, 'UTF-8');
-		$attributes	= array_filter(array($checked, $active, null));
-		$text		= $options[$i]->text;
+		$value      = $value === '' ? '0' : $value;
```
Comment (Quy) [inline layouts/joomla/form/field/radio/switcher.php]: Sorry please move line 72 after line 61 as it should only be evaluated once and not be in the loop. Thanks.

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 14620e006cf05bed8b2f73b47d91ec84dfe60234<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26908">issues.joomla.org/tracker/joomla-cms/26908</a>.</sub>

File: layouts/joomla/form/field/radio/switcher.php (hunk 1 above)
Comment (infograf768) [reply layouts/joomla/form/field/radio/switcher.php]: done. no change in results.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on bf8ee6dd8309e7fe996f42771da7f31ee7266023<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26908">issues.joomla.org/tracker/joomla-cms/26908</a>.</sub>

**Comment by infograf768** [comment]
rtc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26908">issues.joomla.org/tracker/joomla-cms/26908</a>.</sub>

**Comment by Quy** [comment]
Removed RTC to address Harald's comment.

**Comment by alikon** [comment]
what this pr still need to do to became RTC ? unable to find 

> Harald's comment.

**Comment by Quy** [comment]
Discussion in Glip. This PR forces empty string to be 0 which does not allow the case when a value  should be an empty string.

**Comment by brianteeman** [comment]
In what scenario would a radio have an empty string value

**Comment by infograf768** [comment]
@HLeithner 
Please comment.

**Comment by HLeithner** [comment]
@brianteeman for example if the value should be empty^^
```html
<select><option value="">JNONE</option><option value="enlightenment">JENLIGHTENMENT</option></select>
```

For me I don't understand why it should not be possible to have an empty value.

**Comment by SharkyKZ** [comment]
@brianteeman In Global Configuration. Radio fields with 0/1 values are stored as booleans in `configuration.php`. When cast as string `false` becomes an empty string.

**Comment by brianteeman** [comment]
By definition one item in a radio should always have a value unlike a checkbox- or am I completely missing the point here

**Comment by HLeithner** [comment]
An empty value is not invalid.

**Comment by Quy** [comment]
Please test PR #27098.

**Comment by infograf768** [comment]
Shall this be closed?

**Pull request closed without merging**